modules/calendar/calendar_embedder.py

```diff
@@ -122,65 +125,59 @@ async def get_event_choice(
 			allowed_users=[ctx.author],
 		)
 		# get the event selected by the user
-		return events[selection_index]
+		return events_list[selection_index]
 
 	def embed_event_list(
 		self,
 		title: str,
-		events: Generator[Event, None, None],
+		events: peekable[Event],
 		calendar: Calendar,
 		description: str = "",
 		colour: discord.Colour = discord.Colour.blue(),
 		enumeration: Sequence[str] = (),
 		page_num: Optional[int] = None,
 		max_results: int = 10,
-	) -> Tuple[discord.Embed, Generator[Event, None, None]]:
+	) -> discord.Embed:
 		"""Generates an embed with event summaries, links, and dates for each event in the given list
 
 		Arguments:
 
 		:param title: :class:`str` the title to display at the top
-		:param events: :class:`Generator[Event, None, None]` the events to display
+		:param events: :class:`peekable[Event]` the events to display
 		:param calendar: :class:`Calendar` the calendar the events are from
 		:param description: :class:`Optional[str]` the description to embed below the title
 		:param colour: :class:`Optional[discord.Colour]` the embed colour
 		:param enumeration: :class:`Optional[Iterable[str]]` list of emojis to display alongside events (for reaction choices)
 		"""
 		embed = discord.Embed(title=title, colour=colour)
-		# set initial description if available
-		embed.description = "" if description == "" else f"{description}\n"
 		# get calendar links
 		links = self.__calendar_links(calendar)
-		# check if generator is empty
-		empty, events = is_empty(events)
-		if empty:
-			embed.description += "No events found.\n"
-		else:
-			# add events to embed
-			for i in range(max_results):
-				event, events = peek(events)
-				# if event is None, no more events
-				if not event:
-					break
-				# build event description
-				event_description = "\n"
-				# add enumeration emoji if available
-				if i < len(enumeration):
-					event_description += f"{enumeration[i]} "
-				# add the event details
-				event_description += self.__format_event(event)
-				# make sure embed doesn't exceed max size
-				if len(embed.description + event_description + links) > self.max_length:
-					break
-				# add event to embed
-				embed.description += event_description
-				# consume event
-				next(events)
+		# set initial description if available
+		embed.description = f"{description}\n" if description else ""
+		# check if events peekable is empty
+		embed.description += "No events found.\n" if not events else ""
+		# add events to embed
+		for i in range(max_results):
+			next_event = events.peek(None)
+			# if event is None, no more events
+			if not next_event:
+				break
+			# add enumeration emoji if available
```

Comment:
if there are more events to choose from than there are enumerations, theres no point displaying them. we should make `max_results = min(max_results, len(enumeration))` instead of having to check this